cmd/kots/cli/plugin.go

```diff
@@ -0,0 +1,201 @@
+
+	return cmd
+}
+
+func renderAndRunPreflight(preflightFile, configFile string) error {
+	isUrl := true
+	parsed, err := url.ParseRequestURI(preflightFile)
+	if err != nil {
+		isUrl = false
+	}
+
+	var prefBytes []byte
+	if isUrl && parsed != nil && parsed.Scheme == "oci" {
+		// attempt to pull
+		content, err := oci.PullPreflightFromOCI(preflightFile)
+		if err != nil {
+			if err == oci.ErrNoRelease {
+				return errors.Errorf("no release found for %s.\nCheck the oci:// uri for errors or contact the application vendor for support.", preflightFile)
+			}
+
+			return errors.Wrap(err, "failed to pull preflight from oci")
+		}
+
+		prefBytes = content
+	} else {
+		prefBytes, err = ioutil.ReadFile(preflightFile)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to read preflight file: %s", preflightFile))
+		}
+	}
+	var renderedPreflight string
+	renderedPreflight = string(prefBytes)
+
+	// only try to template if a config file is provided
+	if configFile != "" {
+		fb, err := ioutil.ReadFile(configFile)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to read config file: %s", configFile))
+		}
+
+		config, err := kotsutil.LoadConfigFromBytes(fb)
+		if err != nil {
+			return errors.Wrap(err, "failed to load config from bytes")
+		}
+
+		vals := make(map[string]template.ItemValue)
+		builderOptions := template.BuilderOptions{
+			ConfigGroups:    config.Spec.Groups,
+			ExistingValues:  vals,
+			LocalRegistry:   template.LocalRegistry{},
+			License:         nil,
+			Application:     nil,
+			VersionInfo:     nil,
+			ApplicationInfo: nil,
+			IdentityConfig:  nil,
+			Namespace:       "",
+			DecryptValues:   true,
+		}
+
+		builder, _, err := template.NewBuilder(builderOptions)
+		if err != nil {
+			return errors.Wrap(err, "failed to create new config context template builder")
+		}
+
+		renderedPreflight, err = builder.RenderTemplate("preflight", string(prefBytes))
+		if err != nil {
+			return errors.Wrap(err, "failed to render templates in preflight")
+		}
+	}
+
+	preflightSpec, err := kotsutil.LoadPreflightFromContents([]byte(renderedPreflight))
+	if err != nil {
+		return errors.Wrap(err, "failed to load preflight from rendered contents")
+	}
+
+	res, err := preflight.Execute("app", 0, preflightSpec, true, true)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("failed to execute preflights for: %s", preflightSpec.Name))
+	}
+
+	analyzeResults := make([]*analyzerunner.AnalyzeResult, 0)
+	for _, result := range res.Results {
+		res := &analyzerunner.AnalyzeResult{
+			IsPass:  result.IsPass,
+			IsFail:  result.IsFail,
+			IsWarn:  result.IsWarn,
+			Strict:  result.Strict,
+			Title:   result.Title,
+			Message: result.Message,
+			URI:     result.URI,
+		}
+		analyzeResults = append(analyzeResults, res)
+	}
+
+	showStdoutResultsHuman(preflightSpec.Name, analyzeResults)
+	return nil
+}
+
+func showStdoutResultsHuman(preflightName string, analyzeResults []*analyzerunner.AnalyzeResult) error {
```

Comment:
Output should be the normal preflight output as implemented in the troubleshoot project.